src/wallet/rpcwallet.cpp

```diff
@@ -3869,7 +3869,7 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
             nBalance = getBalanceZaddr(addr, std::nullopt, nMinDepth, INT_MAX, false);
         },
         [&](const libzcash::UnifiedAddress& addr) {
-            auto selector = pwalletMain->ZTXOSelectorForAddress(addr, true, false);
+            auto selector = pwalletMain->ZTXOSelectorForAddress(addr, true, false, TransactionStrategy(PrivacyPolicy::FullPrivacy));
```

Comment:
Here we _do_ technically care about this meaning `allowAddressLinkability = false`, because `z_getbalance` returns the balance of an address, and setting `false` here will avoid selecting transparent funds from other addresses.

Reply:
See my footnote in https://github.com/zcash/zcash/pull/6377/files#r1104929323

This call in particular seems to be papering over a bug in `ZTXOSelectorForAddress` – if it’s for an address, why might it return a selector for an account in some cases? And then here we need to pass a parameter to indicate “no, really, it’s for an address”.

And if the rest of my comment with that footnote makes sense, I think that the value passed here should be `AllowRevealedSenders`, otherwise future changes to ZTXOSelectorForAddress to be more strict might break this.